test/api/test_event_client.py

```diff
@@ -0,0 +1,129 @@
+import logging
+import time
+import unittest
+import uuid
+from http import HTTPStatus
+
+import requests
+import responses
+from responses.registries import OrderedRegistry
+
+from devcycle_python_sdk.api.event_client import EventAPIClient
+from devcycle_python_sdk.dvc_options import DevCycleLocalOptions
+from devcycle_python_sdk.exceptions import (
+    APIClientError,
+    APIClientUnauthorizedError,
+    NotFoundError,
+)
+from devcycle_python_sdk.models.event import (
+    UserEventsBatchRecord,
+    RequestEvent,
+    EventType,
+)
+from devcycle_python_sdk.models.user import User
+
+logger = logging.getLogger(__name__)
+
+
+class EventAPIClientTest(unittest.TestCase):
+    def setUp(self) -> None:
+        self.sdk_key = "dvc_server_" + str(uuid.uuid4())
+        options = DevCycleLocalOptions(events_api_uri="http://localhost:8080")
+
+        self.test_client = EventAPIClient(self.sdk_key, options)
+        self.test_batch_url = "http://localhost:8080/v1/events/batch"
+
+        self.test_user = User(user_id="123")
+        events = [
+            RequestEvent(
+                type=EventType.VariableDefaulted,
+                user_id="123",
+                date=time.time(),
+                clientDate=time.time(),
+            )
+        ]
+        self.test_batch = [UserEventsBatchRecord(user=self.test_user, events=events)]
+
+    @responses.activate
+    def test_publish_events(self):
+        responses.add(
+            responses.POST,
+            self.test_batch_url,
+            json={"message": "Successfully received 1 event batches."},
+        )
+
+        message = self.test_client.publish_events(self.test_batch)
+        self.assertEqual(message, "Successfully received 1 event batches.")
```

Comment:
We might want to assert that the body of the request matches the expected format as well here. I think I missed that on the bucketing client as well.

Reply:
good call, i'll add that